### PR TITLE
Create bdd-case-study

### PR DESCRIPTION
**Pull request by ArohiParikh**

### CODE REVIEW

File: content/blog/bdd-case-study.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+image = ""
+draft = false
+authors = ["Incubyte"]
+description = ""
+tags = ["BDD","TDD", "Software Craftsmanship"]
+categories = ["Playbook", "Software Craftsmanship"]
+type = ""
++++
+
+## Implementing Behavior Driven Deployment methodologies to get better Software out the door - a case study
```
Comment (rushaliparikh) [inline content/blog/bdd-case-study.md:14]: Development, not Deployment. Also, check capitalization - don't think Software should be capitalized

File: content/blog/bdd-case-study.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+Over the years, the software development process has seen a massive change of pace. Through ideation, design, development, testing and implementation all the processes involve multiple teams. These groups have varied responsibilities and need to work in a cohesive manner to achieve the end goal.
```
Comment (rushaliparikh) [inline content/blog/bdd-case-study.md:16]:
```suggestion
Over the years, the software development process has seen a massive change of pace. Through ideation, design, development, testing, and implementation, all processes involve multiple skills, which have traditionally been served through multiple teams. These groups have varied responsibilities and need to work in a cohesive manner to achieve the end goal.
```

File: content/blog/bdd-case-study.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+## Bogged down by Waterfall:
```
Comment (rushaliparikh) [inline content/blog/bdd-case-study.md:18]:
```suggestion
## Bogged down by Waterfall
```

File: content/blog/bdd-case-study.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+Fig 1: Communication between multiple distributed teams
+
+Here are some challenges with traditional methodologies:
+·	Vagueness in project requirements due to working in silos as per development process phase
+·	Improper API documentation as the project grows
+·	Treating QA testing as the last process before release
+·	Lack of process automation and resulting slower release cycles
+
+## Focus on behavior and not tools:
```
Comment (rushaliparikh) [inline content/blog/bdd-case-study.md:32]:
```suggestion
## Focus on behavior and not tools
```

File: content/blog/bdd-case-study.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+“A software system can best be designed if the testing is interlaced with the designing, instead of being used after the designing.”
+-	Alan Perils
+
+These days we’re seeing the terms TDD [Test Driven Development, link: https://www.geeksforgeeks.org/test-driven-development-tdd/] and BDD [Behavior Driven Development, link: https://www.techtarget.com/searchsoftwarequality/definition/Behavior-driven-development-BDD] being tossed about a lot, but why the fuss? Radical as it may seem, focus on these processes have become a necessity now if you consider the data from various surveys that show how developers spend almost 1/3rd of their time working to fix issues in their code instead of working on a new feature/enhancement. [source: https://www.businesswire.com/news/home/20210216005484/en/Rollbar-Research-Shows-That-Traditional-Error-Monitoring-Is-Missing-the-Mark]
+
+## TDD helps circumvent these issues by:
```
Comment (rushaliparikh) [inline content/blog/bdd-case-study.md:39]:
```suggestion
**Practices like TDD and BDD help circumvent these issues by:**
```

File: content/blog/bdd-case-study.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+•	Forcing attention on software design from the get-go with a focus on modularity
+•	Promoting an iterative development strategy
+•	Providing a better scalability and maintainability opportunity
+•	Removing the veil of uncertainties involving refactoring code
+
+Using the “shift testing/QA to the left” approach, BDD ensures the essence of what the software is supposed to do, is completely and evenly spread across all vital stakeholders, including QAs and PAs.
```
Comment (rushaliparikh) [inline content/blog/bdd-case-study.md:46]:
```suggestion
Enabling a "fail fast" approach, BDD ensures that the essence of what the software is supposed to do is created in an executable format, bringing in not just automation, but also full clarity around expected behaviour from the get-go.
```

**Review by rushaliparikh — CHANGES_REQUESTED** [review]
When the blog gets into the details of the BDD approach, I think we can do a better job at explaining the process and the benefits. I haven't added line by line suggestions for the last three statements, but it needs work to do justice to the case study.

File: content/blog/bdd-case-study.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+Fig 1: Communication between multiple distributed teams
+
+Here are some challenges with traditional methodologies:
+·	Vagueness in project requirements due to working in silos as per development process phase
+·	Improper API documentation as the project grows
+·	Treating QA testing as the last process before release
+·	Lack of process automation and resulting slower release cycles
+
+## Focus on behavior and not tools
```
Comment (sapanparikh18) [inline content/blog/bdd-case-study.md:32]: I think it should be not THE tool

**Review by sapanparikh18 — CHANGES_REQUESTED** [review]
This blog does not really look like a case study but rather a tutorial on BDD. Needs to be re-written

**Comment by 2KAbhishek** [comment]
Duplicate of #95

**Pull request closed without merging**